libs/auth-nest-tools/src/lib/zendeskAuth.guard.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable, CanActivate, ExecutionContext, Type } from '@nestjs/common'
+import { Request } from 'express'
+import * as crypto from 'crypto'
+
+const SIGNING_SECRET_ALGORITHM = 'sha256'
+
+export function ZendeskAuthGuard(signingSecret: string): Type<CanActivate> {
+  @Injectable()
+  class ZendeskAuthGuardMixin implements CanActivate {
+    canActivate(context: ExecutionContext): boolean {
+      const request: Request = context.switchToHttp().getRequest()
+
+      const signature = request.headers['x-zendesk-webhook-signature'] as string
+      const timestamp = request.headers[
+        'x-zendesk-webhook-signature-timestamp'
+      ] as string
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-expect-error
+      const body = request.rawBody?.toString() ?? ''
```

Comment:
We should maybe create this interface that coderabbit suggests in our infra-nest-server so we can explicitly type this and fix the typescript error instead of ignoring it.